sci-libs/coinor-clp/coinor-clp-1.17.6-r1.ebuild

```diff
@@ -0,0 +1,141 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs
+
+MY_PN=Clp
+
+DESCRIPTION="COIN-OR linear programming solver"
+HOMEPAGE="https://github.com/coin-or/Clp"
+SRC_URI="https://github.com/coin-or/Clp/archive/releases/${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${MY_PN}-releases-${PV}/${MY_PN}"
+
+LICENSE="EPL-1.0"
+SLOT="0/1" # major soname component
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc examples glpk metis mumps sparse static-libs test"
+
+# Fortran is NOT needed, but the ./configure scripts for all of the CoinOR
+# packages contain a check for it. Gentoo bug 601648 and upstream issue,
+#
+#   https://github.com/coin-or/CoinUtils/issues/132
+#
+BDEPEND="
+	virtual/fortran
+	virtual/pkgconfig
+	doc? ( app-doc/doxygen[dot] )
+	test? ( sci-libs/coinor-sample )
+"
+DEPEND="
+	sci-libs/coinor-osi:=
+	sci-libs/coinor-utils:=
+	glpk? ( sci-mathematics/glpk:= sci-libs/amd )
+	metis? ( sci-libs/metis )
+	mumps? ( sci-libs/mumps )
+	sparse? ( sci-libs/cholmod )
+"
+RDEPEND="${DEPEND}"
+
+RESTRICT="!test? ( test )"
+
+src_prepare() {
+	# Needed to make the --with-coin-instdir in src_configure happy.
+	dodir /usr
+
+	mumpslibs="-lmumps_common -ldmumps -lzmumps -lsmumps -lcmumps"
+
+	if has_version sci-libs/mumps[-mpi]; then
```

Comment:
if you flip `USE=mpi` on `sci-libs/mumps`, then this package will break?

Reply:
I've tried to test this thing in a clean chroot and I discovered that mumps is severely underlinked. We need to fix it before trying again

Reply:
If you change the mpi use on mumps you also need to rebuild coinor-clp.
Can this be achieved by using `mumps:=`?

Reply:
USE-based subslot rebuilds been discussed but aren't a thing right now.

Haven't looked at this package's situation, but the idea would generally be to add a USE=mpi, and depend on `mumps[mpi?]`, then check for `use mpi` rather than do has_version checks.

Reply:
Or actually, I think you may need mumps[mpi=] to enforce match.

Reply:
> add a USE=mpi, and depend on `mumps[mpi?]`, then check for `use mpi` rather than do has_version checks.

I like this idea